src/metpy/calc/boundarylayer.py

```diff
@@ -0,0 +1,305 @@
+    See [VH96, Sei00, Col14, Guo16].
+
+    Parameters
+    ----------
+    height : `pint.Quantity`
+        Altitude (metres above ground) of the points in the profile
+    potential_temperature : `pint.Quantity`
+        Potential temperature profile
+    u : `pint.Quantity`
+        Zonal wind profile
+    v : `pint.Quantity`
+        Meridional wind profile
+    smoothingspan : int, optional
+        The amount of smoothing (number of points in moving average)
+    idxfoot : int, optional
+        The index of the foot point (first trusted measure), defaults to 0.
+    bri_threshold : `pint.Quantity`, optional
+        Threshold to exceed to get boundary layer top. Defaults to 0.25
+    ustar : `pint.Quantity`, optional
+        Additional friction term in [VH96]. Defaluts to 0.
+
+    Returns
+    -------
+    blh : `pint.Quantity`
+        Boundary layer height estimation
+    """
+    bRi = bulk_richardson_number(
+        height,
+        smooth(potential_temperature, smoothingspan),
+        smooth(u, smoothingspan),
+        smooth(v, smoothingspan),
+        idxfoot=idxfoot,
+        ustar=ustar,
+    )
+
+    height = height[~np.isnan(bRi)]
+    bRi = bRi[~np.isnan(bRi)]
+
+    if any(bRi > bri_threshold):
+        iblh = np.where(bRi > bri_threshold)[0][0]
+        blh = height[iblh]
+    else:
+        blh = np.nan * units.meter
+
+    return blh
+
+
+def blh_from_parcel(
+    height,
+    potential_temperature,
+    smoothingspan: int = 5,
+    theta0=None,
+):
+    """Calculate atmospheric boundary layer height with the parcel method.
+
+    It is the height where the potential temperature profile reaches its
+    foot value.
+    See [Sei00, HL06, Col14].
+
+    Parameters
+    ----------
+    height : `pint.Quantity`
+        Altitude (metres above ground) of the points in the profile
+    potential_temperature : `pint.Quantity`
+        Potential temperature profile
+    smoothingspan : int, optional
+        The amount of smoothing (number of points in moving average)
+    theta0 : `pint.Quantity`, optional
+        Value of theta at the foot point (skip unstruted points or add extra term). If not provided, theta[0] is taken.
+
+    Returns
+    -------
+    blh : `pint.Quantity`
+        Boundary layer height estimation
+    """
+    potential_temperature = smooth(potential_temperature, smoothingspan)
+
+    if theta0 is None:
+        theta0 = potential_temperature[0]
+
+    if any(potential_temperature > theta0):
+        iblh = np.where(potential_temperature > theta0)[0][0]
+        blh = height[iblh]
+    else:
+        blh = np.nan * units.meter
+
+    return blh
+
+
+def blh_from_humidity_gradient(
+    height,
+    humidity,
+    smoothingspan: int = 5,
+    idxfoot: int = 0,
+):
+    """Calculate atmospheric boundary layer height from the relative
+    humidity gradient
+
+    It is the height where the relative humidity or specific humidity gradient reaches a minimum.
```

Comment:
Definitely possible to use it with TKE or aerosol backscatter. I don't mention it because the references I give are related to radiosounding data, while aerosol would come from lidar or ceilometer and TKE usually from model output, but they are very good way of deriving the BLH (especially the TKE, when turbulence is right in the model).

The wavelet method is out of scope for this first PR, yes. It would be great to have it in the future but I will not have the time to commit for that. In the meantime, this gradient method + smoothing should be equivalent to the Haar wavelet.

Reply:
Sure, Doppler lidars can provide TKE but, as far as I remember, only the most powerful (and expensive) Doppler lidars will be able to have any signal from above the boundary layer, which make it difficult to run such an estimation because the top of the BL might be too close to the end of the profile.

Anyway, the function is there, no harm trying it on TKE profiles! Do you think I should change the name (and the doc) of the function to make it more general? Something like `blh_from_concentration_gradient`?

Reply:
> Sure, Doppler lidars can provide TKE but, as far as I remember, only the most powerful (and expensive) Doppler lidars will be able to have any signal from above the boundary layer, which make it difficult to run such an estimation because the top of the BL might be too close to the end of the profile.

I think I've seen estimates based on SNR/signal quality in that situation, but likely less frequent of a use-case.
> 
> Anyway, the function is there, no harm trying it on TKE profiles! Do you think I should change the name (and the doc) of the function to make it more general? Something like `blh_from_concentration_gradient`?

I like `blh_from_concentration_gradient`.  Definitely mention humidity in the docs, since that's probably what people will be looking for.